tests/integration/helpers.py

```diff
@@ -251,6 +267,37 @@ async def wait_for(
     raise TimeoutError()
 
 
+async def generate_client_from_application(ops_test, jenkins_unit, address) -> UnitWebClient:
+    """Generate a Jenkins client directly from the Juju unit, without fixtures.
+
+    Args:
+        ops_test: OpsTest framework
+        jenkins_unit: Juju Jenkins-k8s unit.
+        address: IP address from the Jenkins unit.
+
+    Returns:
+        A Jenkins web client.
+    """
+    # pylint: disable=duplicate-code
```

Comment:
What part of the code is this duplicated from please?

Reply:
The code here is similar to the one used for the jenkins_client fixture in the integration conftest.py file

Reply:
Perhaps this could be unified?